src/forms/TextInput.jsx

```diff
@@ -5,18 +5,14 @@ import cx from 'classnames';
 /**
  * @module TextInput
  */
-class TextInput extends React.Component {
+class TextInput extends React.PureComponent {
 
 	constructor(props) {
 		super(props);
-		this.state = {
-			value: props.value || '',
-		};
 		this.onChange = this.onChange.bind(this);
 	}
 
 	onChange(e) {
-		this.setState({ value: e.target.value });
 		if (this.props.onChange) {
 			this.props.onChange(e);
```

Comment:
> I think my only concern is how this change would interact if we wanted this component to work with redux forms. @mikespencer any concerns there?

Ah right - so this won't work *as is* with redux-form. [Redux-form passes in props in a specific way](http://redux-form.com/6.8.0/docs/api/Field.md/#props) to the rendered component. I like the idea of having this component being purely presentational though with no ties to a specific library, like redux-form.

One way I'd love to see this work would be to wrap this in another component that **is** meant for consumption by redux-form, roughly something like:

```jsx
import TextInput from 'meetup-web-components';

const ReduxFormTextInput = function ReduxFormTextInput(props) {
  const translatedProps = {
    // convert the props that redux-form passes into this component into something that TextInput understands
  }
  return <TextInput {...translatedProps} />;
}
```

That way, this TextInput can be used in different situations purely for presentation and is totally decoupled from redux-form